### DISCUSSION TITLE
How to transpile ES6 imports in TypeScript

### DISCUSSION THREAD

**Discussion by Cubxity** [swc-project/swc#3527] [Q&A]
:wave: Hey! I got a Node.JS + TypeScript project that I would like to transpile with SWC. My goal is to make the output an ES6 module. 

The issue is that I am not able to import files or directories without explicitly specifying .js/index.js like commonjs or with tsc.

Example code:
```typescript
import B from "./b";
```

I would expect this to automatically suffix `.js` or `/index.js` (if it's a directory), like:
```javascript
import B from "./b.js";

// or..
import B from "./b/index.js"
```

In the current state, the transpiled code would emit the following errors:
```
Error [ERR_UNSUPPORTED_DIR_IMPORT]: Directory import '...' is not supported resolving ES modules imported from ...
```
```
Error [ERR_MODULE_NOT_FOUND]: Cannot find module '...' imported from ...
```

Here is my current `.swcrc`:
```json
{
  "jsc": {
    "parser": {
      "syntax": "typescript",
      "decorators": true
    }
  },
  "module": {
    "type": "es6"
  },
  "env": {
    "targets": {
      "node": 16
    }
  }
}
```

**Comment by Austaras** [comment]


**Reply by Cubxity** [reply]
I've already read that, doesn't seem like it helps with my imports.

**Reply by Austaras** [reply]
> I've already read that, doesn't seem like it helps with my imports.

You should set module type to commonjs, that field is the type of output.

**Reply by Cubxity** [reply]
> > I've already read that, doesn't seem like it helps with my imports.
> 
> You should set module type to commonjs, that field is the type of output.

My goal is to make the output an es6 module.

**Reply by Austaras** [reply]
Sorry I misread your question.

**Reply by mariobm** [reply]
I have a same problem and I don't care about module types, even if it's set to "CommonJS", it does not work.

**Reply by kdy1** [reply]
@mariobm  It's `commonjs`, not `CommonJS`

**Comment by Lionad-Morotar** [comment]
same question here

**Reply by carlbrn** [reply]
In .swcrc add in module `
  "module": {
    "resolveFully": true
  }
`. It will add `/index.js` for the import from the same package.

Although, I still haven't found how to get it to resolve fully when the file is from another package in a monorepo i.e.:
Getting this `import { util } from '@any/pack/helper' => import { util } from '../../pack/helper'` 
Instead of `import { util } from '@any/pack/helper' => import { util } from '../../pack/helper.js'`

**Reply by kdy1** [reply]
@carlbrn Can you file an issue with a minimal reproduction?
